### ISSUE TITLE
Troubles getting binding:env:object:secret annotation to work

### ISSUE THREAD

**Issue by durera** [redhat-developer/service-binding-operator#646]
CRD currently looks something like this (experimenting with a minimal CR that holds mongodb connection properties that I'm trying to use servicebindingrequest to get into a simple demo app):

in namespace 1 we have the CR, and a secret:

The CRD:
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  ...
  annotations:
    servicebindingoperator.redhat.io/status.configDb: binding:env:attribute
    servicebindingoperator.redhat.io/status.credentials-password: binding:env:object:secret
    servicebindingoperator.redhat.io/status.credentials-username: binding:env:object:secret
    servicebindingoperator.redhat.io/status.hosts: binding:env:attribute
  ...
```

The status object of the CR looks like this:
```
status:
  conditions:
  - ansibleResult:
      changed: 2
      completion: 2020-08-27T17:54:00.33111
      failures: 0
      ok: 8
      skipped: 0
    lastTransitionTime: 2020-08-27T17:53:36Z
    message: Awaiting next reconciliation
    reason: Successful
    status: "True"
    type: Running
  configDb: admin
  credentials: secretname
  hosts:
  - host: <removed>
    port: <removed>
  - host: <removed>
    port: <removed>
```

Where `secretname` is a secret that contains the name of a secret that hold username/password in it's data.  

Over in namespace 2, all of the `binding:env:attribute` stuff "just works" -- the servicebindingrequest created and injected the data into my deployment using the intermediary secret:

```
apiVersion: v1
data:
  MONGODB_CONFIGDB: <removed>
  MONGODB_HOSTS_0_HOST: <removed>
  MONGODB_HOSTS_0_PORT: <removed>
  MONGODB_HOSTS_1_HOST: <removed>
  MONGODB_HOSTS_1_PORT: <removed>
kind: Secret
...
```

But I cannot for the life of me get the username and password pulled out of the referenced secret.  

The docs seem to indicate that the annotation should be `servicebindingoperator.redhat.io/status.credentials.password: binding:env:object:secret`, however looking at the postgres example the format used there has a "-" to seperate the resource and the property name, implying I should be using an annotation of `servicebindingoperator.redhat.io/status.credentials-password: binding:env:object:secret` (see: [PR](https://github.com/operator-backing-service-samples/postgresql-operator/commit/2d1821c1d4ebb43940e44e9f6f75af48aadd66e4). 

Neither works for me so far, and I can't see any errors in the service binding operator logs either, and the status of the binding request indicates it's happy enough.

```
  - lastHeartbeatTime: 2020-08-27T18:34:30Z
    lastTransitionTime: 2020-08-27T18:34:14Z
    status: "True"
    type: CollectionReady
  - lastHeartbeatTime: 2020-08-27T18:34:30Z
    lastTransitionTime: 2020-08-27T18:34:14Z
    status: "True"
    type: InjectionReady
```

Kind of at a loss where to go from here to be honest :)

**Comment by sbose78** [comment]
Your annotations look right off the top of my head. Looking further 🤔


 The docs PR got merged a while back coincidentally https://github.com/redhat-developer/service-binding-operator/blob/master/docs/BackingServiceBestPractices.md

**Comment by sbose78** [comment]
Does `secretname`  have the user and password?

**Comment by durera** [comment]
> Does `secretname` have the user and password?

Thanks, after looking at the secret to check this, I realised where I'd gone wrong -- there was no `ownerReferences` in there to tie the secret to the CR.  Doh!

Using this approach https://sdk.operatorframework.io/docs/building-operators/ansible/reference/retroactively-owned-resources/, the operator now retroactively claims ownership of the secret and the whole things works as expected!

I don't know whether I missed an error somewhere in the logs about this, or whether there isn't one ... but rather than the `CollectionReady` status going to True, would it be possible to flag some kind of error under that condition in a situation like this, or is the srb operator unable to know there's a problem in this scenario?

```
  - lastHeartbeatTime: 2020-08-27T18:34:30Z
    lastTransitionTime: 2020-08-27T18:34:14Z
    message: "Secret xxx doesn't exist / key xxx not found in secret xxx data/etc"
    status: "False"
    type: CollectionReady
``

**Comment by sbose78** [comment]
The logs should not have royally ignored the issue 🤷‍♂️
Curious, What does your full ServiceBindingRequest look like?

**Comment by DhritiShikhar** [comment]
Thanks for the good feedback @durera 

+1 to adding information about this in status.

**Comment by sbose78** [comment]
@pedjak @isutton could you see why the annotations didn't work here ? They look fine to me.

**Comment by sbose78** [comment]
Also, wonder if the following is documented 🤔   I had a very hard time finding it the other day :D  

```
  MONGODB_HOSTS_0_HOST: <removed>
  MONGODB_HOSTS_0_PORT: <removed>
  MONGODB_HOSTS_1_HOST: <removed>
  MONGODB_HOSTS_1_PORT: <removed>
```

**Comment by pedjak** [comment]
It should be checked and eventually fixed in #661

**Comment by pratikjagrut** [comment]
Test case scenario got merged #787 closing this issue.
